clar2wasm/tests/wasm-generation/contracts.rs

```diff
@@ -159,3 +159,80 @@ proptest! {
         crosscheck(&format!("(as-contract {value})"), Ok(Some(value.into())));
     }
 }
+
+proptest! {
+    #![proptest_config(super::runtime_config())]
+
+    #[test]
+    fn contract_dynamic_call_accepts_any_args(
+        (tys, values)
+            in prop::collection::vec(
+                prop_signature().prop_ind_flat_map2(|ty| PropValue::from_type(ty.clone())),
+                1..=20
+            )
+            .prop_map(|arg_ty| arg_ty.into_iter().unzip::<_, _, Vec<_>, Vec<_>>())
+            .no_shrink(),
+        result in PropValue::any().no_shrink(),
+        err_type in prop_signature(),
+    ) {
+        // first contract
+        let first_contract_name = "foo".into();
+        let mut function_types = String::new();
+        let mut function_arguments = String::new();
+        for (name, ty) in ('a'..).zip(tys.iter()) {
+            let ty = type_string(ty);
+            write!(function_arguments, "({name} {ty}) ").unwrap();
+            function_types += &(ty + " ");
+        }
+        let first_snippet = format!(
+            r#"
+                (define-trait foo-trait (
+                    (foofun ({function_types}) (response {} {}))
+                ))
+
+                (define-public (foofun {function_arguments})
+                    (ok {result})
+                )
+            "#,
+            result.type_string(),
+            type_string(&err_type)
+        );
+
+        // second contract
+        let second_contract_name = "bar".into();
+
+        let contract_call_args: String =
+            ('a'..)
```

Comment:
This is cool. I didn't know you could do that.

Reply:
Old haskeller habbit to work with lazy infinite structures :)